include/net/net_if.h

```diff
@@ -1516,7 +1516,7 @@ static inline void net_if_ipv4_set_gw(struct net_if *iface,
  * will return the default network interface if the best network interface
  * is not found.
  */
-struct net_if *net_if_select_src_iface(struct sockaddr *dst);
+struct net_if *net_if_select_src_iface(const struct sockaddr *dst);
```

Comment:
Please consider moving this change to the previous commit (which introduced the function declaration), to ensure bisectability ;-).